lib/src/barback/asset_environment.dart

```diff
@@ -618,6 +628,89 @@ class AssetEnvironment {
     });
   }
 
+  bool __paused = false;
+  bool get _paused => __paused;
+  void set _paused(bool newState) {
+    __paused = newState;
+    if (_paused) {
+      if (_scheduledUpdate != null) {
+        _scheduledUpdate.cancel();
+        _scheduledUpdate = null;
+        _resumeServers();
+      }
+    } else {
+      _doUpdate();
+    }
+  }
+
+  void _pauseServers() {
+    for (var server in _servers.values) {
+      server.pause();
+    }
+  }
+
+  void _resumeServers() {
+    for (var server in _servers.values) {
+      server.resume();
+    }
+  }
+
+  /// An update that is scheduled to happen sometime in the future, or `null`.
+  CancelableOperation _scheduledUpdate;
+
+  /// Schedules a cancellable update for [_buildDelay]ms in the future.
+  ///
+  /// If one is already scheduled then it is cancelled, and this one takes its
+  /// place.
+  void _scheduleUpdate() {
+    if (_paused) return;
+    if (_buildDelay == 0) {
+      _doUpdate();
+      return;
+    }
+
+    if (_scheduledUpdate == null) _pauseServers();
+    _scheduledUpdate?.cancel();
+    var cancelled = false;
+    _scheduledUpdate = new CancelableOperation.fromFuture(
+        new Future.delayed(new Duration(milliseconds: _buildDelay), () {
+          if (cancelled) return;
+          _scheduledUpdate = null;
+          _doUpdate();
+          _resumeServers();
+        }), onCancel: () {
+      cancelled = true;
+    });
```

Comment:
It would probably be cleaner to use a [`RestartableTimer`](https://www.dartdocs.org/documentation/async/1.13.3/async/RestartableTimer-class.html) here.